app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class CartsController < ApplicationController
+  
+  before_action :set_cart
+
+  def add_to_cart
+    @product = Product.find(params[:product_id])
+    @line_item = @cart.line_items.find_by(product: @product)
+
+    if @line_item
+      @line_item.increment(:quantity)
+    else
+      @line_item = @cart.line_items.build(product: @product, quantity: 1)
+    end
```

Comment:
Let's make this a one liner like:-

@line_item = @line_item ? @line_item.increment(:quantity) : @cart.line_items.build(product: @product, quantity: 1)